File: server/internal/entity/summary.go
package entity

import "time"

type Summary struct {
	SummaryID int       `gorm:"primaryKey;autoIncrement:true" json:"summaryID"`
	UserID    int       `gorm:"not null" json:"userID"`
	Title     string    `gorm:"type:varchar(255);not null" json:"title"`
	Content   string    `gorm:"type:text;not null" json:"content"`
	CreatedAt time.Time `gorm:"type:timestamp;default:CURRENT_TIMESTAMP" json:"createdAt"`
	UpdatedAt time.Time `gorm:"type:timestamp;default:CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP" json:"updatedAt"`
}

File: client/lib/model/summary.dart
class Summary {
  final int summaryId;
  final int userId;
  final String title;
  final String content;
  final DateTime createdAt;
  final DateTime updatedAt;

  Summary({
    required this.summaryId,
    required this.userId,
    required this.title,
    required this.content,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Summary.fromJson(Map<String, dynamic> json) {
    return Summary(
      summaryId: json['summaryid'],
      userId: json['userid'],
      title: json['title'],
      content: json['content'],
      createdAt: DateTime.parse(json['createdat']),
      updatedAt: DateTime.parse(json['updatedat']),
    );
  }
}
